nmigen/lib/fifo.py

```diff
@@ -366,6 +371,25 @@ def elaborate(self, platform):
             self.r_rdy.eq(~r_empty),
         ]
 
+        # Reset handling to maintain FIFO and CDC invariants in the presence of a write-domain
+        # reset. For further discussion, see https://github.com/nmigen/nmigen/issues/181.
```

Comment:
Do you think you could summarize it here? You have all the relevant context right now, whereas that issue is fairly long and meandering. In cases like this I normally describe the problem, the possible solution, and the rationale for choosing the one I implemented, in a comment near the code. This way, anyone who reads the source will know exactly why it works the way it does, without having to parse an entire discussion.